### PR TITLE
Add new rule: `require-form-method`

### PR DESCRIPTION
**Pull request by raido**
This PR adds new rule for form elements to require method attribute with value `POST/post` or `GET/get`.

Closes: https://github.com/ember-template-lint/ember-template-lint/issues/1203

### CODE REVIEW

**Comment by rwjblue** [comment]
Adding a rule for this seems totally fine, but I'm trying to decide if we should eventually try to get this into the default rule set (after a major bump).

@Turbo87 / @bmish / @dcyriller - Curious what y'all's thoughts are here?

**Comment by bmish** [comment]
I thought there were many situations where `get` forms are completely acceptable, like search results `?query=my-search-term`.

File: test/unit/rules/require-form-method-post-test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  config: true,
+
+  good: [
+    '<form method="POST"></form>',
+    '<form method="post"></form>',
+
+    // dynamic values
+    '<form method="{{formMethod}}"></form>',
+    '<form method={{formMethod}}></form>',
+  ],
```
Comment (bmish) [inline test/unit/rules/require-form-method-post-test.js]: Need to add `good` test cases to ensure false positives are avoided such as:

* `<div></div>`
* `<div method="get"></div>`

File: docs/rule/require-form-method-post.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# require-form-method-post
+
+This rule requires all `<form>` elements to have `method` attribute with `POST` value.
+
+By default `form` elements without `method` attribute are submitted as `GET` requests.
+In usual applications `submit` event listeners are attached to `form` elements and `event.preventDefault()` is called to avoid form submission.
+
+However in case of failure to prevent default action, form submission as `GET` request can leak sensitive end-user information.
```
Comment (bmish) [inline docs/rule/require-form-method-post.md]: This should mention that there are acceptable use cases for `GET` as well: https://www.w3schools.com/tags/att_form_method.asp

**Comment by Turbo87** [comment]
I tend to agree with @bmish. I think converting this to `require-form-method` without enforcing the `post` value would be better, potentially with a `allow-get: false` option if necessary.

**Comment by raido** [comment]
I’ll convert it to generic rule with config option for disabling GET. I thought about it when implementing this as well. But still went ahead with just post to get something out for feedback.

**Comment by raido** [comment]
This is ready for review and has been converted to generic rule with configurability.

File: lib/rules/require-form-method.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  parseConfig(config) {
+    let configType = typeof config;
+
+    switch (configType) {
+      case 'boolean':
+        return config ? DEFAULT_CONFIG : false;
+      case 'object':
+        if (Array.isArray(config.allowedMethods)) {
+          return {
+            allowedMethods: config.allowedMethods.map((m) => String(m).toUpperCase()),
```
Comment (bmish) [inline lib/rules/require-form-method.js]: Can you throw an exception if any unexpected methods are passed in?

File: docs/rule/require-form-method.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## Configuration
+
+The following values are valid configuration:
+
+* boolean - `true` to enable / `false` to disable
+* object -- An object with the following keys:
+  * `allowedMethods` -- An array of allowed form `method` attribute values, default: `['POST', 'GET']`
+
+## References
```
Comment (bmish) [inline docs/rule/require-form-method.md]: This is a good, simple link too: https://www.w3schools.com/tags/att_form_method.asp

File: docs/rule/require-form-method.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# require-form-method
+
+This rule requires all `<form>` elements to have `method` attribute with `POST` or `GET` value.
+
+By default `form` elements without `method` attribute are submitted as `GET` requests.
+In usual applications `submit` event listeners are attached to `form` elements and `event.preventDefault()` is called to avoid form submission.
+
+However in case of failure to prevent default action, form submission as `GET` request can leak sensitive end-user information.
+Examples of valid uses for `GET` requests are non-secure data, bookmarking the submission result, data search query strings.
```
Comment (bmish) [inline docs/rule/require-form-method.md]: I would prefer bullet-point list instead of a sentence here to make it easier to parse the acceptable usages.

Comment (raido) [reply docs/rule/require-form-method.md]: Converted to list.

File: docs/rule/require-form-method.md (hunk 4 above)
Comment (raido) [reply docs/rule/require-form-method.md]: Added.

File: lib/rules/require-form-method.js (hunk 3 above)
Comment (raido) [reply lib/rules/require-form-method.js]: Added. I also added "dialog" method type as by HTML spec to defaults list.

File: lib/rules/require-form-method.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+        if (tag !== 'form') {
+          return;
+        }
+
+        let typeAttribute = attributes.find((it) => it.name === 'method');
+        if (!typeAttribute) {
+          this.logNode({
+            node,
+            message: ERROR_MESSAGE.replace('%', this.config.allowedMethods),
```
[the remark this answers is not in the document]
Comment (raido) [reply lib/rules/require-form-method.js]: Done.

**Comment by raido** [comment]
I need to push a fix after breaking out of the loop. It does not break the switch and throw an exception in case invalid methods are configured as input. Missing conditional there for switch break.

File: docs/rule/require-form-method.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+The following values are valid configuration:
+
+* boolean - `true` to enable / `false` to disable
+* object -- An object with the following keys:
+  * `allowedMethods` -- An array of allowed form `method` attribute values, default: `['POST', 'GET', 'DIALOG']`
+
+## References
+
+* [W3 Schools - form method attribute](https://www.w3schools.com/tags/att_form_method.asp)
```
Comment (mehulkar) [inline docs/rule/require-form-method.md:46]: Could link to mdn instead: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-method

Comment (raido) [reply docs/rule/require-form-method.md:46]: I can replace that with MDN.

Comment (raido) [reply docs/rule/require-form-method.md]: Done.

**Comment by dcyriller** [comment]
Thank you!

**Pull request merged**